### PR TITLE
[PFS] Make pfs try start with fixed port.

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Start: always start from a fixed port
      While port is used:
            Port +=1
      Dump port to port file
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py [hunk 1]
```diff
@@ -125,10 +126,15 @@ def is_port_in_use(port: int):
         return s.connect_ex(("localhost", port)) == 0
 
 
-def get_random_port():
-    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-        s.bind(("localhost", 0))
-        return s.getsockname()[1]
+def get_pfs_port():
+    port = 23333
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:130]: : ) I think we should add some docs on this constant port in our pf service start help

Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:130]: added

File: docs/reference/pf-command-reference.md [hunk 2]
```diff
@@ -905,7 +905,8 @@ pf service start [--port]
 ```
 
 #### Examples
-Start prompt flow service.
+Start prompt flow service. We endeavor to start the service on the default port 23333. And if it's already taken, we will sequentially probe new ports, incrementing by one each time.
```
Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:908]: Avoid use "we"

We endeavor to start -> Promptflow will stry to start

Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:908]: document here and also the pf service start --help that will remember and use the last started port if port not set

Comment (YingChen1996) [reply docs/reference/pf-command-reference.md:908]: we will not use the last started port for now, but always start from default port or usered passed port.

when start service: always start from 23333

when stop service: will clear port info

Comment (YingChen1996) [reply docs/reference/pf-command-reference.md:908]: offline synced and updated

**Review by Adrian999999999 — APPROVED** [review]

File: docs/reference/pf-command-reference.md [hunk 3]
```diff
@@ -905,18 +905,22 @@ pf service start [--port]
 ```
 
 #### Examples
-Start prompt flow service. We endeavor to start the service on the default port 23333. And if it's already taken, we will sequentially probe new ports, incrementing by one each time.
+Prompt flow will try to start the service on the default port 23333 if "pfs.port" file doesn't exist. If the port is
```
Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:908]: Don't mention pfs.port file. User will not understand what's this

File: docs/reference/pf-command-reference.md [hunk 4]
```diff
@@ -905,18 +905,22 @@ pf service start [--port]
...
+Prompt flow will try to start the service on the default port 23333 if "pfs.port" file doesn't exist. If the port is 
+already taken, prompt flow will sequentially probe new ports, incrementing by one each time. If the port file exists, 
+prompt flow will just try to start the service on the port number specified in the file.
 
 ```bash
 pf service start
 ```
 
-Force restart promptflow service.
+Forcefully restart the promptflow service even if the port is already in use.
```
Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:916]: we don't always restart. rephrase it like: 

Forcefully start the promptflow service. If the port is already in use, xxx

File: docs/reference/pf-command-reference.md [hunk 5]
```diff
@@ -905,18 +905,22 @@ pf service start [--port]
...
 ```bash
 pf service start --force
 ```
 
-Start promptflow service with specific port.
+Start the promptflow service with a specified port. If the port is already taken, prompt flow will raise an error 
+unless forcefully start the service with the `--force` flag. Upon availability, the port number will be logged in the 
+"pfs.port" file.
```
Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:923]: don't say pfs.port

File: src/promptflow-devkit/promptflow/_cli/_pf/_service.py [hunk 6]
```diff
@@ -93,7 +92,11 @@ def add_parser_start_service(subparsers):
     pf service start --port 65553
     """  # noqa: E501
     add_param_port = lambda parser: parser.add_argument(  # noqa: E731
-        "-p", "--port", type=int, help="port of the prompt flow service"
+        "-p",
+        "--port",
+        type=int,
+        help="The designated port of the prompt flow service and port number will be logged "
+        "in the pfs.port file if port is available.",
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_pf/_service.py:99]: mark pfs.port file

**Review by wangchao1230 — DISMISSED** [review]

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-devkit/promptflow/_cli/_pf/_service.py [hunk 7]
```diff
@@ -91,7 +92,10 @@ def add_parser_start_service(subparsers):
     pf service start --port 65553
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_cli/_pf/_service.py:92]: change to 65535，
65553 is  most likely not working

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py [hunk 8]
```diff
@@ -99,13 +96,18 @@ def get_port_from_config(create_if_not_exists=False):
...
     if is_run_from_built_binary():
         port_file_path = HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE
         port_file_path.touch(mode=read_write_by_user(), exist_ok=True)
     else:
-        # Set port to ~/.promptflow/pfs/**_pf.port, if already have a port in file , will overwrite it.
         port_file_path = get_current_env_pfs_file(PF_SERVICE_PORT_FILE)
+    return port_file_path
+
+
+def dump_port_to_config(port):
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:108]: should we make the port as a Configuration item?

Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:108]: we will always start from PF_SERVICE_DEFAULT_PORT now

**Review by D-W- — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**